sonar-db/src/main/java/org/sonar/db/DefaultDatabase.java

```diff
@@ -82,15 +86,14 @@ void initSettings() {
     properties.setProperty(DatabaseProperties.PROP_DRIVER, dialect.getDefaultDriverClassName());
   }
 
-  private void initDatasource() throws Exception {// NOSONAR this exception is thrown by BasicDataSourceFactory
+  private void initDataSource() throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)